### ISSUE TITLE
Is using a 3rd party promise library still supported?

### ISSUE THREAD

**Issue by mcfarljw** [googleapis/nodejs-datastore#605]
I remember this used to be supported as a DatastoreOption. It was something like:

```
new Datastore({
  promise: require('bluebird')
})
```

I noticed that some other cloud libraries still support it, for example storage:
https://github.com/googleapis/nodejs-storage/blob/9fee6d969a1311a3db18bf523d3614adef0526ce/src/storage.ts#L51

Is this something that was intentionally removed?

**Comment by JustinBeckwith** [comment]
Greetings!  The custom promise support was useful a few years ago when promises weren't widely available, or performant in modern browsers and versions of node.js.  For the last few years though, supported versions of nodejs and all evergreen browsers have really caught up. 

It's possible this still works, but it's really not something we're going to actively support moving forward.  

I'd highly recommend rolling with the out-of-the-box promise implementation :)

**Comment by mcfarljw** [comment]
Thanks @JustinBeckwith that answers my question. We have just removed bluebird from our project to give native promises a go as the primary reason we were not using them before was the performance which as you mentioned is kind of a non-issue these days.

The reason I was wanting to use bluebird was for promise cancellation, but we resolved our specific issue by passing a timeout value to `gaxOptions` so that it throws errors.

Speaking of `gaxOptions` the docs (https://googleapis.github.io/gax-nodejs/interfaces/CallOptions.html) for them are not very useful as they contain no descriptions about those settings. I'm not sure where I'd post about that though.
